// ionic-native/dist/esm/plugins/istudy.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
/**
 * This is a template for new plugin wrappers
 *
 * TODO:
 * - Add/Change information below
 * - Document usage (importing, executing main functionality)
 * - Remove any imports that you are not using
 * - Add this file to /src/index.ts (follow style of other plugins)
 * - Remove all the comments included in this template, EXCEPT the @Plugin wrapper docs.
 * - Remove this note
 *
 */
import { Plugin, Cordova } from './plugin';
/**
 * @name istudy
 * @description
 * This plugin does something
 *
 * @usage
 * ```
 * import { istudy } from 'ionic-native';
 *
 * istudy.functionName('Hello', 123)
 *   .then((something: any) => doSomething(something))
 *   .catch((error: any) => console.log(error));
 *
 * ```
 */
var Istudy = (function () {
    function Istudy() {
    }
    /**
     * This function does something
     * @param arg {any} Some param to configure something
     * @return {Promise<any>} Returns a promise that resolves when something happens
     */
    Istudy.getSecrityCode = function (arg) {
        return; // We add return; here to avoid any IDE / Compiler errors
    };
    Istudy.getHeader = function (arg) {
        return; // We add return; here to avoid any IDE / Compiler errors
    };
    Istudy.encrypt = function (arg) {
        return;
    };
    Istudy.verify = function (arg) {
        return;
    };
    Istudy.generateMd5 = function (arg) {
        return;
    };
    Istudy.getDeviceInfo = function (arg) {
        return;
    };
    return Istudy;
}());
__decorate([
    Cordova({
        successIndex: 0,
        errorIndex: 1
    })
], Istudy, "getSecrityCode", null);
__decorate([
    Cordova({
        successIndex: 0,
        errorIndex: 1
    })
], Istudy, "getHeader", null);
__decorate([
    Cordova({
        successIndex: 0,
        errorIndex: 1
    })
], Istudy, "encrypt", null);
__decorate([
    Cordova({
        successIndex: 0,
        errorIndex: 1
    })
], Istudy, "verify", null);
__decorate([
    Cordova({
        successIndex: 0,
        errorIndex: 1
    })
], Istudy, "generateMd5", null);
__decorate([
    Cordova({
        successIndex: 0,
        errorIndex: 1
    })
], Istudy, "getDeviceInfo", null);
Istudy = __decorate([
    Plugin({
        pluginName: 'Istudy',
        plugin: 'cordova-plugin-istudy',
        pluginRef: 'Istudy',
        repo: '',
        install: '' // OPTIONAL install command, in case the plugin requires variables
    })
], Istudy);
export { Istudy };
