// ionic-native/dist/es5/plugins/testp.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
/**
 * This is a template for new plugin wrappers
 *
 * TODO:
 * - Add/Change information below
 * - Document usage (importing, executing main functionality)
 * - Remove any imports that you are not using
 * - Add this file to /src/index.ts (follow style of other plugins)
 * - Remove all the comments included in this template, EXCEPT the @Plugin wrapper docs.
 * - Remove this note
 *
 */
var plugin_1 = require("./plugin");
// import { Observable } from 'rxjs/Observable';
/**
 * @name testp
 * @description
 * This plugin does something
 *
 * @usage
 * ```
 * import { testp } from 'ionic-native';
 *
 * testp.functionName('Hello', 123)
 *   .then((something: any) => doSomething(something))
 *   .catch((error: any) => console.log(error));
 *
 * ```
 */
var Testp = (function () {
    function Testp() {
    }
    /**
     * This function does something
     * @param arg1 {string} Some param to configure something
     * @param arg2 {number} Another param to configure something
     * @return {Promise<any>} Returns a promise that resolves when something happens
     */
    Testp.tp1 = function () {
        // return;
    };
    return Testp;
}());
__decorate([
    plugin_1.Cordova()
], Testp, "tp1", null);
Testp = __decorate([
    plugin_1.Plugin({
        pluginName: 'Testp',
        plugin: 'cordova-plugin-testp',
        pluginRef: 'Testp',
        repo: '',
        install: '' // OPTIONAL install command, in case the plugin requires variables
    })
], Testp);
exports.Testp = Testp;
